libs/sdk-ui-dashboard/src/tools/InsightWidgetBuilder.ts

```diff
@@ -21,51 +21,74 @@ import {
  * @internal
  */
 export class InsightWidgetBuilder {
-    insight: ObjRef;
-    type: string = "insight";
-    ignoredDashboardFilters: IDashboardFilterReference[] = [];
-    drills: DrillDefinition[] = [];
-    title: string = "";
-    description: string = "";
-    configuration: IInsightWidgetConfiguration = { hideTitle: false };
-    properties: VisualizationProperties = {};
+    widget: IInsightWidgetBase = {
+        insight: { uri: "" },
+        type: "insight",
+        ignoreDashboardFilters: [],
+        drills: [],
+        title: "",
+        description: "",
+        configuration: { hideTitle: false },
+        properties: {},
+    };
 
     constructor(insight: IInsight) {
-        this.insight = insightRef(insight);
-        this.title = insightTitle(insight);
+        this.widget = {
+            ...this.widget,
```

Comment:
I don't think the spreads are necessary, why not just mutate the object and spread only in the `build()`?